p2p/rebroadcast.go

```diff
@@ -153,21 +156,24 @@ func (r *Rebroadcast) RegainMempool() {
 }
 
 func (r *Rebroadcast) onRegainMempool() {
-	mptxCount := r.s.TxMemPool().Count()
-	if !r.regainMP {
-		if mptxCount > 0 {
-			return
-		}
-	}
 	if !r.s.PeerSync().IsCurrent() {
 		return
 	}
+	mptxCount := r.s.TxMemPool().Count()
 
-	r.regainMP = false
-
+	canPeers := []*peers.Peer{}
 	for _, pe := range r.s.Peers().CanSyncPeers() {
-		go r.s.sy.Send(pe, synch.RPCMemPool, &pb.MemPoolRequest{TxsNum: uint64(mptxCount)})
+		if time.Since(pe.GetMempoolReqTime()) <= params.ActiveNetParams.TargetTimePerBlock {
+			continue
+		}
+		canPeers = append(canPeers, pe)
+	}
+	if len(canPeers) <= 0 {
+		return
 	}
+	index := rand.Int() % len(canPeers)
+	pe := canPeers[index]
```

Comment:
1. better to use `Intn()`
```
rand.Intn(len(canPeers))
```
2. the `math.rand` is auto seeded only after go1.20, the prior code results a fixed index. see
https://pkg.go.dev/math/rand#Seed

Reply:
OK